scripts/check-criterion-output.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# check-criterion-output.sh
+#
+# SUMMARY
+#
+#   Used to reformat benchmark output from criterion into a condensed format. It
+#   reads from stdin and writes to stdout.
+
+DIR="$(dirname "${BASH_SOURCE[0]}")"
+
+awk --file "$DIR/parse-criterion-output.awk" |
+  jq --slurp --raw-output '.[] | [.name, .time, .time_change, .throughput, .throughput_change, .change] | @tsv' |
+  column --separator $'\t' --table --table-columns "name,time,time change,throughput,throughput change,change" |
+  awk -v rc=0 '/regressed/ { rc=1 } 1; END { if (rc == 1) { print "\nRegression detected. Note that any regressions should be verified."; exit rc }}'
```

Comment:
Could we set a tolerance level? Benchmarking has notoriously high variance, especially on public CI boxes.

Reply:
I'm hoping to fine tune that by configuring the `noise_threshold` for the benchmark groups: https://docs.rs/criterion/0.3.3/criterion/struct.BenchmarkGroup.html#method.noise_threshold so that Criterion takes them into account when calculating. I wanted to see what kind of noise we have though, first. I'm also exploring having these run on self-hosted runners.